### PR TITLE
Pay with Business accounts 

### PR DESCRIPTION
**Pull request by OmerGery**
image

### CODE REVIEW

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/FutureRides/StopPointRow.js [hunk 1]
```diff
@@ -1,7 +1,5 @@
 import React from 'react';
-import { View } from 'react-native';
 import styled from 'styled-components';
-import moment from 'moment';
```
Comment (OmerGery) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/FutureRides/StopPointRow.js:4]: unuseds

File: examples/client/Locomotion/src/Components/TabSwitch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+color: #666975;
+color: ${({ isSelected }: TabStyled) => (isSelected ? SELECTED_COLOR : UNSELECTED_COLOR)};
+font-family: Montserrat;
+font-size: 16px;
+font-style: normal;
+font-weight: 500;
+line-height: 24px;
+padding-left: 4px;
+`;
+
```
Comment (ormiz) [inline examples/client/Locomotion/src/Components/TabSwitch/index.tsx]: move to styles file

File: examples/client/Locomotion/src/Components/BusinessAccountText/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+font-size: 14px;
+font-style: normal;
+line-height: 20px;
+`;
+const SubTitle = styled(BaseText)`
+font-weight: 500;
+`;
+const BoldTitle = styled(BaseText)`
+font-weight: 700;
+`;
```
Comment (ormiz) [inline examples/client/Locomotion/src/Components/BusinessAccountText/index.tsx]: Move to styles file

**Review by ormiz — APPROVED** [review]

File: examples/client/Locomotion/src/Components/BusinessAccountText/styled.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+export const TitleWithSubTitle = styled(View)`
+  display: flex;
+  flex-direction: column;
+  flex: 1 0 0;
+`;
+export const BaseText = styled(Text)`
+color: #212229;
+font-family: Inter;
+font-size: 14px;
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/BusinessAccountText/styled.ts]: use theme context

File: examples/client/Locomotion/src/Components/CardRow/index.tsx [hunk 5]
```diff
@@ -49,7 +51,7 @@ const margin = `margin-${Start()}`;
 const TextContainer = styled(View)`
   justify-content: center;
   ${margin}: 16px;
-  width: 80%;
+  width: ${({ noSvg } : { noSvg: boolean}) => (noSvg ? '95%' : '80%')};
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/CardRow/index.tsx]: Use flex

File: examples/client/Locomotion/src/Components/CardRow/index.tsx [hunk 6]
```diff
@@ -205,6 +232,17 @@ const CardRow = (paymentMethod: any) => {
             {paymentMethod.disabledReason}
           </Description>
           )}
+          {(paymentMethod.mark && paymentMethod.alignMarkToRight) ? (
+            <SvgIcon
+              style={{
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/CardRow/index.tsx:237]: Why it needs custom style? you may use flex to place it right

File: examples/client/Locomotion/src/Components/EmptyState/styled.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+font-size: 14px;
+font-style: normal;
+font-weight: 600;
+`;
+export const Description = styled(Text)`
+align-self: stretch;
+color: #666975;
+text-align: center;
+font-family: Inter;
+font-size: 14px;
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/EmptyState/styled.ts]: use theme context

File: examples/client/Locomotion/src/Components/EmptyState/styled.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+margin: 4px;
+flex: 1 0 0;
+align-self: stretch;
+`;
+export const Title = styled(Text)`
+align-self: stretch;
+color: #212229;
+text-align: center;
+font-family: Inter;
+font-size: 14px;
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/Components/EmptyState/styled.ts]: use theme context

**Review by ormiz — APPROVED** [review]

File: examples/client/Locomotion/src/context/newRideContext/index.tsx [hunk 9]
```diff
@@ -258,17 +267,21 @@ const RidePageContextProvider = ({ children }: {
...
   const stopRequestInterval = () => {
     clearInterval(intervalRef.current);
   };
-
   const saveLastRide = async (rideId: string) => {
     await StorageService.save({ lastRideId: rideId });
   };
+  const saveOrderedRidePaymentMethod = async (rideBusinessAccountId: string | null) => Promise.all([
+    StorageService.save({ lastBusinessAccountId: rideBusinessAccountId || PAYMENT_MODES.PERSONAL }),
+    StorageService.save({ orderedRide: true }),
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/newRideContext/index.tsx:282]: merge the payment method logic

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/RideButtons/index.tsx [hunk 10]
```diff
@@ -83,9 +85,10 @@ const RideButtons = ({
   const [passengersCounterError, setPassengersCounterError] = useState(false);
   const firstDate = () => moment(ride?.scheduledTo || undefined).add(ride?.scheduledTo ? 0 : (minMinutesBeforeFutureRide || 0) + 1, 'minutes').toDate();
   const [tempSelectedDate, setTempSelectedDate] = useState(firstDate());
-
-  const paymentMethodNotAllowedOnService = chosenService && ride?.paymentMethodId
-    && !chosenService.allowedPaymentMethods.includes(getPaymentMethod(ride.paymentMethodId));
+  const paymentMethodNotAllowedOnService = !businessAccountId
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/RideButtons/index.tsx]: maybe split it to 2 variables? will be easier to read

File: examples/client/Locomotion/src/popups/ChoosePaymentMethod/styled.js [hunk 11]
```diff
@@ -43,10 +43,13 @@ export const SummaryContainer = styled.View`
 `;
 
 export const Title = styled.Text`
-  ${FONT_SIZES.H2}
-  color: black;
-  ${FONT_WEIGHTS.SEMI_BOLD};
   flex: 10;
+  color: #212229;
+  font-family: Inter;
+  font-size: 16px;
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/popups/ChoosePaymentMethod/styled.js]: use theme context

File: examples/client/Locomotion/src/popups/ChoosePaymentMethod/index.tsx [hunk 12]
```diff
@@ -61,27 +78,78 @@ const PaymentMethodPopup = ({
   };
 
   useEffect(() => {
-    usePayments.getOrFetchCustomer();
+    usePayments.loadCustomer();
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/popups/ChoosePaymentMethod/index.tsx:81]: why?

Comment (OmerGery) [reply examples/client/Locomotion/src/popups/ChoosePaymentMethod/index.tsx:81]: resolving, discussed

File: examples/client/Locomotion/src/Components/CardRow/index.tsx (hunk 5 above)
Comment (OmerGery) [reply examples/client/Locomotion/src/Components/CardRow/index.tsx]: undid the changes, it actually doesnt affect anything

**Pull request merged**